File: python/semantic_kernel/ai/open_ai/services/open_ai_config.py
# Copyright (c) Microsoft. All rights reserved.

from typing import Optional


# TODO: allow for overriding endpoints
class OpenAIConfig:
    """
    The OpenAI configuration.
    """

    # OpenAI model name, see https://platform.openai.com/docs/models
    model_id: str
    # OpenAI API key, see https://platform.openai.com/account/api-keys
    api_key: str
    # OpenAI organization ID. This is usually optional unless your
    # account belongs to multiple organizations.
    org_id: Optional[str]

    def __init__(
        self, model_id: str, api_key: str, org_id: Optional[str] = None
    ) -> None:
        """Initializes a new instance of the OpenAIConfig class.

        Arguments:
            model_id {str} -- OpenAI model name, see
                https://platform.openai.com/docs/models
            api_key {str} -- OpenAI API key, see
                https://platform.openai.com/account/api-keys
            org_id {Optional[str]} -- OpenAI organization ID.
                This is usually optional unless your
                account belongs to multiple organizations.
        """
        if not model_id:
            raise ValueError("The model ID cannot be empty")
        if not api_key:
            raise ValueError("The API key cannot be empty")

        self.model_id = model_id
        self.api_key = api_key
        self.org_id = org_id

File: dotnet/src/Connectors/Connectors.OpenAI/OpenAIConfig.cs
// Copyright (c) Microsoft. All rights reserved.

using Microsoft.SemanticKernel.Diagnostics;

namespace Microsoft.SemanticKernel.Connectors.OpenAI;

/// <summary>
/// OpenAI configuration.
/// </summary>
public sealed class OpenAIConfig : ServiceConfig
{
    /// <summary>
    /// OpenAI model name, see https://platform.openai.com/docs/models
    /// </summary>
    public string ModelId { get; }

    /// <summary>
    /// OpenAI API key, see https://platform.openai.com/account/api-keys
    /// </summary>
    public string APIKey { get; }

    /// <summary>
    /// OpenAI organization id. This is usually optional unless your account belongs to multiple organizations.
    /// </summary>
    public string? OrgId { get; }

    /// <summary>
    /// Creates a new <see cref="OpenAIConfig" /> with supplied values.
    /// </summary>
    /// <param name="serviceId">An identifier used to map semantic functions to AI services and models,
    /// decoupling prompts configurations from the actual model used.</param>
    /// <param name="modelId">OpenAI model name, see https://platform.openai.com/docs/models</param>
    /// <param name="apiKey">OpenAI API key, see https://platform.openai.com/account/api-keys</param>
    /// <param name="orgId">OpenAI organization id. This is usually optional unless your account belongs to multiple organizations.</param>
    public OpenAIConfig(string serviceId, string modelId, string apiKey, string? orgId)
        : base(serviceId)
    {
        Verify.NotEmpty(modelId, "The model ID is empty");
        Verify.NotEmpty(apiKey, "The API key is empty");

        this.ModelId = modelId;
        this.APIKey = apiKey;
        this.OrgId = orgId;
    }
}
